vespalib/src/vespa/vespalib/net/tls/impl/openssl_tls_context_impl.cpp

```diff
@@ -137,7 +137,10 @@ X509Ptr read_untrusted_x509_from_bio(::BIO& bio) {
 
 ::SSL_CTX* new_tls1_2_ctx_with_auto_init() {
     ensure_openssl_initialized_once();
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
     return ::SSL_CTX_new(::TLSv1_2_method());
+#pragma GCC diagnostic pop
```

Comment:
Out of curiosity, what warning is emitted by GCC here?

Reply:
Changing ignored to error in the pragma on line 141 gives the following error:

/home/tegge/git/vespa/vespalib/src/vespa/vespalib/net/tls/impl/openssl_tls_context_impl.cpp: In function ‘SSL_CTX* vespalib::net::tls::impl::{anonymous}::new_tls1_2_ctx_with_auto_init()’:
/home/tegge/git/vespa/vespalib/src/vespa/vespalib/net/tls/impl/openssl_tls_context_impl.cpp:142:28: error: ‘const SSL_METHOD* TLSv1_2_method()’ is deprecated [-Werror=deprecated-declarations]
     return ::SSL_CTX_new(::TLSv1_2_method());
                            ^~~~~~~~~~~~~~

diagnostic pop is just to restore diagnostic state to before the last diagnostic push (or command line defined diagnostic state if no matching diagnostic push).